src/Refactoring-Core/RBExtractToTemporaryRefactoring.class.st

```diff
@@ -42,14 +43,13 @@ RBExtractToTemporaryRefactoring class >> model: aRBSmalltalk extract: anInterval
 RBExtractToTemporaryRefactoring >> applicabilityPreconditions [
 
 	^ {
-		  (RBCondition definesSelector: selector in: class).
-		  (RBCondition
-			   isValidInstanceVariableName: newVariableName
-			   for: class).
-		  (RBCondition withBlock: [
-			   self verifySelectedInterval.
-			   self checkVariableName.
-			   true ]) }
+		  (ReDefinesSelectorsCondition new definesSelectors: (Array with: selector) in: class).
+		  (ReIsValidInstanceVariableName new name: newVariableName).
+		  (ReBlockCondition new 
+				block: [
+					self verifySelectedInterval.
+			   		self checkVariableName.
```

Comment:
Would it make sense to create condition classes for these two conditions as well?
- `verifySelectedInterval` just checks if the selection is a sequence,
- `checkVariableName` should be already defined as a condition.
I suggest at least to inline `verifySelectedInterval` and add `ReCheckVariableName` to the list of preconditions.